keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -796,33 +860,72 @@ class ClusterHandler(
       }
     }
 
-  private suspend fun CloudDriverService.getServerGroups(resource: Resource<ClusterSpec>): Iterable<ServerGroup> =
-    getServerGroups(
+  private suspend fun CloudDriverService.getActiveServerGroups(resource: Resource<ClusterSpec>): Iterable<ServerGroup> {
+    val existingServerGroups: Map<String, List<ClouddriverServerGroup>> = getExistingServerGroupsByRegion(resource)
+    val activeServerGroups = getActiveServerGroups(
       account = resource.spec.locations.account,
       moniker = resource.spec.moniker,
       regions = resource.spec.locations.regions.map { it.name }.toSet(),
       serviceAccount = resource.serviceAccount
-    )
-      .also { them ->
-        val allSame: Boolean = them.distinctBy { it.launchConfiguration.appVersion }.size == 1
-        val healthy: Boolean = them.all {
-          it.instanceCounts?.isHealthy(resource.spec.deployWith.health) == true
-        }
-        if (allSame && healthy) {
-          // // only publish a successfully deployed event if the server group is healthy
-          val appVersion = them.first().launchConfiguration.appVersion
-          if (appVersion != null) {
-            publisher.publishEvent(
-              ArtifactVersionDeployed(
-                resourceId = resource.id,
-                artifactVersion = appVersion
-              )
-            )
-          }
+    ).map { activeServerGroup ->
+      // get all server groups in the cluster, split by region,
```

Comment:
This comment doesn't seem to match the code below, since the result is just a flat list. Maybe "get all active server groups in the cluster and tag the ones that are the only enabled ones in their region"? Also, maybe move the comment to line 865?

Reply:
removing the comment! good call. it was left over / out of place